macros/contexts/contextNondecimalBase.pl

```diff
@@ -0,0 +1,268 @@
+
+sub new {
+	my $self    = shift;
+	my $class   = ref($self) || $self;
+	my $context = bless Parser::Context->getCopy('Numeric'), $class;
+	$context->{name}           = 'NondecimalBase';
+	$context->{parser}{Number} = 'context::NondecimalBase::Number';
+	$context->{value}{Real}    = 'context::NondecimalBase::Real';
+	$context->functions->disable('All');
+	$context->operators->undefine('/');
+	$context->constants->clear();
+	$context->{precedence}{NondecimalBase} = $context->{precedence}{special};
+	$context->flags->set(limits => [ -1000, 1000, 1 ]);
+	return $context;
+}
+
+# set the base of the context.  Either base a number >=2 or an arrayref of digits.
+sub setBase {
+	my ($self, $base) = @_;
+	my $digits;
+
+	if (ref($base) eq 'ARRAY') {
+		$digits = $base;
+		$base   = scalar(@$digits);
+		die 'Base must be at least 2' unless $base >= 2;
+	} else {
+		die 'Base must be an integer' unless $base == int($base);
+		die 'Base must be at least 2' unless $base >= 2;
+		die 'You must provide a digit list for bases bigger than 36' if $base > 36;
+		$digits = [ ('0' .. '9', 'A' .. 'Z')[ 0 .. $base - 1 ] ];
+	}
+
+	$self->{base}            = $base;
+	$self->{digits}          = $digits;
+	$self->{digitMap}        = { map { ($digits->[$_], $_) } (0 .. $base - 1) };
+	$self->{pattern}{number} = '[' . join('', @$digits) . ']+';
+	$self->update;
+}
+
+sub copy {
+	my $self = shift;
+	my $copy = $self->SUPER::copy;
+	$copy->{base}     = $self->{base};
+	$copy->{digits}   = $self->{digits};
+	$copy->{digitMap} = $self->{digitMap};
+	return $copy;
+}
+
+# Convert a number in base10 to the given base.
+sub toBase {
+	my ($self, $base10) = @_;
+	my $b      = $self->{base};
+	my $digits = $self->{digits};
+
+	my @baseB;
+	do {
+		my $d = $base10 % $b;
+		$base10 = ($base10 - $d) / $b;
+		unshift(@baseB, $digits->[$d]);
+	} while $base10;
+
+	return join('', @baseB);
+}
+
+# Convert a number in a given base to base 10.
+sub fromBase {
+	my ($self, $baseB) = @_;
+	my $b      = $self->{base};
+	my $digits = $self->{digits};
+	my $digit  = $self->{digitMap};
+
+	my $base10 = 0;
+	for my $d (split('', $baseB)) {
+		die 'The number to convert must consist only of digits: ' . join(',', @$digits) unless defined($digit->{$d});
+		$base10 = $base10 * $b + $digit->{$d};
+	}
+
+	return $base10;
+}
+
+# A replacement for Parser::Number that acepts numbers in a non-decimal base and
+# converts them to decimal for internal use
+package context::NondecimalBase::Number;
+our @ISA = ('Parser::Number');
+
+# Create a new number in the given base and convert to base 10.
+sub new {
+	my ($self, $equation, $value, $ref) = @_;
+	my $context = $equation->{context};
+
+	Value::Error('The base must be set for this context') unless $context->{base};
+
+	$value = $context->fromBase($value);
+	return $self->SUPER::new($equation, $value, $ref);
+}
+
+sub eval {
+	$self = shift;
+	my $base = $self->{equation}{context}{base};
```

Comment:
Since `$base` is never used, this line should be removed.

Reply:
fixed.